File: MoveZeroes(283).js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let startIdx = -1;
    let lastIdx = -1;
    
    for(let i=0; i<nums.length; i++) {
        if(nums[i]==0) {
            if(startIdx==-1) {
                startIdx = i;
                endIdx = startIdx;
            } else {
                endIdx = i;
            }
        } else if(startIdx != -1) {
            nums[startIdx] = nums[i];
            nums[i] = 0;
            if(startIdx!=endIdx){
                startIdx += 1;
                endIdx = i;
            } else {
                startIdx = i;
                endIdx = startIdx;
            }   
        }
    }
    
    return nums;
};

File: MoveZeroes(283).swift
class Solution {
	func moveZeroes(_ nums: inout [Int]) {
		var zeroIdx = -1
		
		for i in 0..<nums.count {
			if nums[i] == 0 {
				if zeroIdx == -1 {
					zeroIdx = i
				}
			} else if zeroIdx != -1 {
				nums[zeroIdx] = nums[i]
				nums[i] = 0
				zeroIdx += 1
			}
		}
	}
}
